packages/orbit-components/src/primitives/ButtonPrimitive/index.js

```diff
@@ -50,6 +51,12 @@ export const StyledButtonPrimitive = styled(
       const Component = isButtonWithHref ? "a" : asComponent;
       const buttonType = submit ? "submit" : "button";
 
+      const handleKeyDown = () => {
```

Comment:
Having the `handleKeyDown` won't work – it can't be a function in this particular case. It doesn't have any parameter, even though it should have – to pass down the event.
```js
const handleKeyDown = ev => {
  if (asComponent === "div") {
    return onKeyDown(onClick)(ev);
  }
  return null;
};
```

Better would be to have have it as:
```js
const handleKeyDown = asComponent === "div" ? onKeyDown(onClick) : null;
```